File: 1874.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().split('\n')

const TESTCASE = input[0]
let temp = []
let stack = []
let count = 0
let result = []
let answer = ''

for (let i = 0; i < TESTCASE; i++) {
    temp[i] = i + 1
}

for (let i = 1; i <= TESTCASE; i++) {
    count = 1

    while (count <= TESTCASE && stack[stack.length - 1] != input[i]) {
        stack.push(temp[0])
        temp.shift()
        result.push('+')
        count++
    }

    if (stack[stack.length - 1] == input[i]) {
        stack.pop()
        result.push('-')
    } else {
        console.log('NO')
        return
    }
}

for (let i = 0; i < result.length; i++) {
    answer += result[i] + '\n'
}

console.log(answer)

File: Baekjoon/python/stack/1874.py
testcase = int(input())
temp = 1
stack = []
count = 0
result = []
answer = ''

for i in range(1, testcase+1):
    count = 1
    n = int(input())
    while count <= testcase:
        if stack:
            if stack[-1] == n:
                break
        stack.append(temp)
        result.append('+')
        count += 1
        temp += 1
    if stack[-1] == n:
        stack.pop()
        result.append('-')
    else:
        print('NO')
        exit()
for i in result:
    answer += str(i) + '\n'
print(answer)
